wrapper/src/main/java/software/amazon/jdbc/plugin/cache/CacheConnection.java

```diff
@@ -277,6 +289,48 @@ public CacheConnection(final Properties properties) {
     this(properties, null);
   }
 
+  /**
+   * Detects whether the Redis endpoint is running in cluster mode by executing INFO command.
+   * Caches the result to avoid repeated detection. The caller of this function needs to hold a lock for thread safety.
+   */
+  private void detectClusterMode() {
+    if (isClusterMode != null) {
+      return;
+    }
+
+    String[] hostnameAndPort = getHostnameAndPort(cacheRwServerAddr);
+    RedisURI redisUri = buildRedisURI(hostnameAndPort[0], Integer.parseInt(hostnameAndPort[1]));
+
+    ClientResources resources = ClientResources.builder().build();
+
+    try (RedisClient client = RedisClient.create(resources, redisUri);
+         StatefulRedisConnection<byte[], byte[]> conn = client.connect(new ByteArrayCodec())) {
+
+      String infoOutput = conn.sync().info("cluster");
+      boolean clusterEnabled = false;
+      if (infoOutput != null) {
+        // Parse the INFO output line by line
+        for (String line : infoOutput.split("\r?\n")) {
+          if (line.startsWith("cluster_enabled:")) {
+            String value = line.substring("cluster_enabled:".length()).trim();
+            clusterEnabled = "1".equals(value);
+            break;
+          }
+        }
+      }
+      isClusterMode = clusterEnabled;
+      LOGGER.info("Detected cache mode: " + (clusterEnabled ? "CLUSTER" : "STANDALONE"));
```

Comment:
Would be good to also log the cluster endpoint DNS + port here for debugging purposes.